File: PesanRahasia/PesanRahasia.php
<?php
/**
* Algoritma 
* 1. Buat array berisi kamus ROT13  
* 2. Baca baris input.8
* 3. Pecah baris ke dalam potongan karakter
* 4. Cocokkan karakter pada kolom "key" array, jika ketemu ambil "value" dari "key" tersebut
* 5. Cocokkan karakter pada kolom "value" array, jika ketemu ambil "key"-nya   
* 6. Jika langkah 4 dan 5 tidak memberikan hasil, biarkan karakter tersebut apa adanya
* 7. Gabungkan hasil pada langkah 4,5, dan 6 hingga membentuk kalimat yg 
     sudah ter-decrypt. Cetak ke layar !
* 8. Kembali ke langkah 2, terus ulangi hingga baris terakhir selesai dibaca
**/

class PesanRahasia
{
    protected $baskom;
    protected $hasilTerjemah = "";
    protected $kamus = array("a" => "n", "b" => "o", "c" => "p", "d" => "q",
                             "e" => "r", "f" => "s", "g" => "t", "h" => "u",
                             "i" => "v", "j" => "w", "k" => "x", "l" => "y",
                             "m" => "z");

    public function bacaSandi()
    {
        $pencacah = 0; 
        try { 
        	$handle = @fopen("input.8", "r");
            while (($buffer = fgets($handle, 4096)) !== false) {
                if ($pencacah != 0) {
                    // Terjemahkan Baris 
		    		$this->baskom = str_split($buffer);
                    foreach ($this->baskom as $key => $value) {
                        $this->hasilTerjemah .= $this->terjemahkan($value);
                    }
                    echo $pencacah . "). " . $this->hasilTerjemah;
                    $this->hasilTerjemah = "";
                }
                $pencacah++;
            } 
            fclose($handle);
         } catch (Exception $ex) {
         	echo $ex->getMessage();
         }
    }

    private function terjemahkan($karakter)
    {
        if (array_key_exists($karakter, $this->kamus)) {
            return $this->kamus[$karakter];
        } else {
            if ($kunci = array_search($karakter, $this->kamus)) {
                return $kunci;
            } else {
                return $karakter;
            }
        }
    }

}
$PesanRahasia = new PesanRahasia();
$PesanRahasia->bacaSandi();
?>

File: PesanRahasia/PesanRahasia.py
#!/usr/bin/python -tt
import sys

class PesanRahasia:

  kamus = {"a": "n", "b": "o", "c": "p", "d": "q",
           "e": "r", "f": "s", "g": "t", "h": "u",
           "i": "v", "j": "w", "k": "x", "l": "y",
           "m": "z"}

  def bongkar(self):
    hasil = ''
    pembaca = open('input.8', 'r')
    n = int(pembaca.readline()) 
    for i in range(1, n+1):
      baris = pembaca.readline().strip()
      potonganHuruf = list(baris) 
      for huruf in potonganHuruf:
        hasil += self.terjemahkan(huruf)
      print 'Case #%d: %s' % (i, hasil)+'\n'
      hasil = ''    

  def terjemahkan(self,karakter):
    ketemu = 0
    # Indeks pencarian = key
    for key,values in self.kamus.iteritems():
      if karakter == key:
        terjemahanHuruf = values
        ketemu = 1
    
    # Indeks pencarian = values
    for key,values in self.kamus.iteritems():
      if karakter == values:
        terjemahanHuruf = key
        ketemu = 1
    
    # Kesimpulan hasil pencarian
    if ketemu == 1:
      return terjemahanHuruf
    else:
      return karakter

    
PesanRahasia = PesanRahasia()
PesanRahasia.bongkar()
